### PR TITLE
adding Signup Functionality

### LINKED ISSUE #84: Add Signup Functionality
Signup screen needs to be designed and implemented.Signup feature is implemented in android.

Would you like to work on this issue?
Yes

### PR DESCRIPTION
**Pull request by SIMRAN1**
Fixes #84
Signup Functionality is Added .Now User can Signup through the app

### CODE REVIEW

**Comment by SIMRAN1** [comment]
@imjog @chashmeetsingh please review!! @imjog I have solved problem of ScrollView in this PR and tested in simulator!

File: FOSSAsia/AuthTabBarViewController.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class AuthTabBarViewController: UITabBarController {
+
+    @IBOutlet weak var backButton: UIBarButtonItem!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.parent?.navigationItem.backBarButtonItem = nil
+
+        let button = UIButton(type: .custom)
```
Comment (jogendra) [inline FOSSAsia/AuthTabBarViewController.swift]: Please use functions and put code inside them and call here. Do not write code in `viewDidLoad`.

**Review by jogendra — CHANGES_REQUESTED** [review]
- On Signup screen, the spinner is always visible. Only show spinner when it is needed, otherwise keep it hidden.
- Show email field line green only when email is valid, otherwise keeps it red.
- Signup screen is not in `scrollView`.

**Comment by SIMRAN1** [comment]
okk @imjog !

**Comment by SIMRAN1** [comment]
Here is the link for demonstration https://drive.google.com/open?id=1hhCXRgWuaXsBYkHVWN4_gtYB7zGCgJWV

**Comment by SIMRAN1** [comment]
@chashmeetsingh @nikit19 @dreadpool2 @simarsingh24 @imjog please review!

**Comment by jogendra** [comment]
@SIMRAN1 In the video demonstration, it is seen that when the user types their last name, the last name text filed is hiding behind the keyboard. that's what I was pointing the last PR and in this PR. This need to be solved. thanks

**Comment by SIMRAN1** [comment]
ok @imjog I will solve it!

**Comment by SIMRAN1** [comment]
Done @imjog !

**Review by jogendra — APPROVED** [review]
LGTM :)

**Review by chashmeetsingh — APPROVED** [review]

**Pull request merged**